conn.go

```diff
@@ -457,3 +497,18 @@ func (l *Conn) reader() {
 		}
 	}
 }
+
+// a variation of net.SplitHostPort which returns string as port if not specified.
+func splitHostAndPort(hostport string) (string, string, error) {
+
+	if hostport == "" {
+		return "", "", errors.New("Empty hostport provided")
+	}
+
+	// make golang < 1.8 compatible
+	i := strings.LastIndexByte(hostport, ':')
+	if i == -1 || i < strings.IndexByte(hostport, ']') {
```

Comment:
this doesn't look proper... why is this needed?

Reply:
this test checks that collon is after braces, e,g, to exclude IPv6 address without port, like **[::1]**

Reply:
I think we should simply use SplitHostPort and assume the input is either a valid host or host:port

Reply:
SplitHostPort returns an error when port is not specified. This makes it hard to identify invalid string from missing port, unless we want to parse the error.

Reply:
we shouldn't try to identify an invalid string... if url.Parse passes, we should accept the host blob. if SplitHostPort fails, we should assume it is a port-less host

Reply:
probably will cover the majority of usecases.... :+1: